packages/bundler/src/modules/BundleManager.ts

```diff
@@ -131,19 +131,40 @@ export class BundleManager {
       } = parsedError.args
       const userOp = userOps[opIndex]
       const reasonStr: string = reason.toString()
+
+      // find entity to blame
```

Comment:
Please extract this to a function and call it `findEntityToBlameForThisFailure` or something expressive like that.